src/class-wpml-gutenberg-integration.php

```diff
@@ -80,32 +81,44 @@ private function register_blocks( array $blocks, array $package_data ) {
 
 		foreach ( $blocks as $block ) {
 
-			if ( ! $block instanceof WP_Block_Parser_Block && $block['blockName'] ) {
-				$block = new WP_Block_Parser_Block( $block['blockName'], $block['attrs'], $block['innerBlocks'], $block['innerHTML'], $block['innerContent'] );
-			}
+			$block   = $this->sanitize_block( $block );
+			$strings = $this->strings_in_blocks->find( $block );
 
-			if ( $block instanceof WP_Block_Parser_Block ) {
+			foreach ( $strings as $string ) {
 
-				$strings = $this->strings_in_blocks->find( $block );
+				do_action(
+					'wpml_register_string',
+					$string->value,
+					$string->id,
+					$package_data,
+					$string->name,
+					$string->type
+				);
 
-				foreach ( $strings as $string ) {
+			}
 
-					do_action(
-						'wpml_register_string',
-						$string->value,
-						$string->id,
-						$package_data,
-						$string->name,
-						$string->type
-					);
+			if ( isset( $block->innerBlocks ) ) {
+				$this->register_blocks( $block->innerBlocks, $package_data );
+			}
+		}
+	}
 
-				}
+	/**
+	 * @param WP_Block_Parser_Block|array $block
+	 *
+	 * @return WP_Block_Parser_Block
+	 */
+	private function sanitize_block( $block ) {
```

Comment:
Passing through this method will ensure that we get a `WP_Block_Parser_Block` object and we can process it as such.